File: mobile-client/android/app/src/main/java/app/mobiling/client/auth/AuthFeatureBridge.kt
package app.mobiling.client.auth

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.RegisterAuthRequest
import app.mobiling.client.contract.auth.session.StartAuthRequest
import app.mobiling.client.data.auth.session.AuthSessionGateway
import app.mobiling.client.usecase.auth.session.RegisterAuthUseCase
import app.mobiling.client.usecase.auth.session.StartAuthUseCase

/**
 * Marketing America Corp. Oleksandr Tishchenko
 *
 * App-level bridge for auth entry flow.
 *
 * It keeps Auth as the dedicated entry/auth surface while delegating
 * behavior into canonical Contract/Data/UseCase slices.
 */
class AuthFeatureBridge(
    private val gateway: AuthSessionGateway,
) {
    suspend fun start(request: StartAuthRequest): AuthSessionPayload =
        StartAuthUseCase(gateway).invoke(request)

    suspend fun register(request: RegisterAuthRequest): AuthSessionPayload =
        RegisterAuthUseCase(gateway).invoke(request)
}

File: mobile-client/ios/App/Auth/AuthFeatureBridge.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
// App-level bridge for auth entry flow.
public struct AuthFeatureBridge: Sendable {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func start(request: StartAuthRequest) async throws -> AuthSessionPayload {
        try await StartAuthUseCase(gateway: gateway)(request: request)
    }

    public func register(request: RegisterAuthRequest) async throws -> AuthSessionPayload {
        try await RegisterAuthUseCase(gateway: gateway)(request: request)
    }
}
